src/libcore/benches/iter.rs

```diff
@@ -345,3 +345,13 @@ fn bench_partial_cmp(b: &mut Bencher) {
 fn bench_lt(b: &mut Bencher) {
     b.iter(|| (0..100000).map(black_box).lt((0..100000).map(black_box)))
 }
+
+#[bench]
+fn bench_starts_with_vec(b: &mut Bencher) {
+    b.iter(|| black_box((0..100)).collect::<Vec<_>>().starts_with(&[0..3]));
+}
```

Comment:
As a benchmark baseline you might want to compare to a handrolled loop over two slices while using `get_unchecked`.
A longer shared prefix may also help highlighting the iteration cost.